### PR TITLE
subsystem-in-subject: don't check for merge commits

### PR DESCRIPTION
**Pull request by tmrts**

### CODE REVIEW

File: rules/subsystem/subsystem.go [hunk 1]
```diff
@@ -24,9 +25,16 @@ func init() {
 
 // ValidateSubsystem checks that the commit's subject is of the form '<subsystem>: <commit-msg>'.
 func ValidateSubsystem(c git.CommitEntry) (vr validate.Result) {
+	vr.CommitEntry = c
+	if len(strings.Split(c["parent"], " ")) > 1 {
+		vr.Pass = true
+		vr.Msg = "merge commits do not require subsystem in subject"
```
Comment (jonboulle) [inline rules/subsystem/subsystem.go:31]: Is this necessary? wouldn't we only expect merge commits to happen from the GH merge, which won't show up in the PR?

Comment (jonboulle) [inline rules/subsystem/subsystem.go:31]: i.e. we want a clean log with the exception of GH merges

Comment (tmrts) [inline rules/subsystem/subsystem.go:31]: It's up to you, feel free to close it if you decide we don't need it

Comment (jonboulle) [inline rules/subsystem/subsystem.go:31]: I think it's fine without this after the HEAD..master diff thing is fixed

we still need line 37 :-)

**Comment by tmrts** [comment]
Closing in favor of #2

**Comment by tmrts** [comment]
Re-opened, because this feature prevents headaches, where the PR of a user might not be rebased on the latest HEAD and that might cause the tested range of commits to include merge commits.

**Comment by jonboulle** [comment]
can we check for github merge commits specifically?

**Comment by tmrts** [comment]
@jonboulle what are your concerns? AFAIK there are no commits with multiple parents that are not merge commits.

**Comment by jonboulle** [comment]
I don't want a bunch of 'merge coreos/rkt into master' if people aren't
actually rebasing...

**Comment by tmrts** [comment]
@jonboulle that is an entirely different matter though, isn't it? You want the users to rebase their PR to up-to-date HEAD before submitting the PR. We can do that with repo settings

**Comment by jonboulle** [comment]
eh? How can you do that with repo settings?

**Comment by tmrts** [comment]
@jonboulle if what you want from users is that they rebase their PRs to latest HEAD before pushing a PR, there is an option with the description `Require branches to be up to date before merging`

**Comment by jonboulle** [comment]
@tmrts does that prevent the following?

```
  git checkout master && git pull
  git checkout -b cool_feature_branch # create a branch
  git commit -am "add a new feature"
  git checkout master && git pull && git - # refresh master
  git merge master # merge master in (instead of rebasing) - this is the thing I'm trying to avoid by disallowing merge commits in the PR diff
  git push && git pull-request
```

**Comment by jonboulle** [comment]
Given the speed of CI I don't think we can sanely enable `Require branches to be up to date before merging` unless we get a merge bot :/

**Comment by tmrts** [comment]
@jonboulle no, let's address that as a new rule `no-merge-commits` the responsibility of this rule is different

**Comment by tmrts** [comment]
@jonboulle I misunderstood your use case, enabling that repo option wouldn't be productive

**Comment by jonboulle** [comment]
please file a follow up :-)

**Pull request merged**